### PR TITLE
Addition of Omise in Event Wizard

### LINKED ISSUE #2862: Addition of Omise Gateway in Event Wizard
See Parent Issue -: #2859 

**Additional Context**

Working on it

### PR DESCRIPTION
**Pull request by kushthedude**
#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

#### Short description of what this resolves:

Addition of Omise in `Event Wizard Step`

See Parent Issue #2859 
 

Fixes #2862

### CODE REVIEW

**Comment by kushthedude** [comment]
@uds5501 How does it looks ?

File: app/models/event.js [hunk 1]
```diff
@@ -57,6 +57,7 @@ export default ModelBase.extend(CustomPrimaryKeyMixin, {
   isStripeLinked  : attr('boolean'),
   canPayByCheque  : attr('boolean', { defaultValue: false }),
   canPayByBank    : attr('boolean', { defaultValue: false }),
+  canPayByOmise   : attr('boolean', { defaultValue: false }),
```
Comment (uds5501) [inline app/models/event.js]: @kushthedude Please add corresponding changes in `events.py` schema and model, otherwise it looks good to go!

Comment (kushthedude) [reply app/models/event.js]: @uds5501 ,Will open the PR in Server, Thanks for reminding 😅

Comment (kushthedude) [reply app/models/event.js]: @CosmicCoder96 Review Please

**Comment by kushthedude** [comment]
@CosmicCoder96 Review Please

File: app/templates/components/forms/orders/order-form.hbs [hunk 2]
```diff
@@ -136,6 +136,11 @@
                 <label for="payment_mode"><img src="/images/payment-logos/paypal.png" alt="paypal"></label>
               </div>
             {{/if}}
+            {{#if data.event.canPayByOmise}}
+              {{ui-radio name='payment_mode' label=(t 'Omise') value='Omise' current=data.paymentMode
+                         onChange=(action (mut data.paymentMode))}}
+              <label for="payment_mode"><img src="/images/payment-logos/omise.png" alt="omise"></label>
```
Comment (CosmicCoder96) [inline app/templates/components/forms/orders/order-form.hbs]: If you didn't add an image, how would it appear?

File: app/components/forms/wizard/basic-details-step.js [hunk 3]
```diff
@@ -44,6 +44,10 @@ export default Component.extend(FormMixin, EventWizardMixin, {
     return this.get('settings.isStripeActivated') && find(paymentCurrencies, ['code', this.get('data.event.paymentCurrency')]).stripe;
   }),
 
+  canAcceptOmise: computed('data.event.paymentCurrency', 'settings.isOmiseActivated', function() {
+    return this.get('settings.isOmiseActivated') && find(paymentCurrencies, ['code', this.get('data.event.paymentCurrency')]).omise;
```
Comment (CosmicCoder96) [inline app/components/forms/wizard/basic-details-step.js]:
```js
find(paymentCurrencies, ['code', this.get('data.event.paymentCurrency')]).omise;
```
If you never included omise in currencies, why would this possibly work? Don't make PRs if you haven't tested locally.

File: app/templates/components/forms/wizard/basic-details-step.hbs [hunk 4]
```diff
@@ -308,14 +308,33 @@
           </div>
         {{/if}}
       {{/if}}
+      {{#if canAcceptOmise}}
+        <label>{{t 'Payment with Omise'}}</label>
+        <div class="field payments">
+          <div class="ui checkbox">
+            {{input type='checkbox' id='payment_by_omise' checked=data.event.canPayByOmise}}
+            <label for="payment_by_omise">
+              <span>{{t 'Yes, accept payment through Omise Gatewat'}}</span>
```
Comment (CosmicCoder96) [inline app/templates/components/forms/wizard/basic-details-step.hbs]: Typo.

Comment (CosmicCoder96) [inline app/templates/components/forms/wizard/basic-details-step.hbs]: why the unnecessary span tag?

File: app/templates/components/forms/orders/order-form.hbs [hunk 5]
```diff
@@ -136,6 +136,11 @@
                 <label for="payment_mode"><img src="/images/payment-logos/paypal.png" alt="paypal"></label>
               </div>
             {{/if}}
+            {{#if data.event.canPayByOmise}}
+              {{ui-radio name='payment_mode' label=(t 'Omise') value='Omise' current=data.paymentMode
```
Comment (CosmicCoder96) [inline app/templates/components/forms/orders/order-form.hbs]:
```js
value = 'omise'
```

File: app/templates/components/forms/orders/order-form.hbs [hunk 6]
```diff
@@ -136,6 +136,11 @@
                 <label for="payment_mode"><img src="/images/payment-logos/paypal.png" alt="paypal"></label>
               </div>
             {{/if}}
+            {{#if data.event.canPayByOmise}}
```
Comment (CosmicCoder96) [inline app/templates/components/forms/orders/order-form.hbs]: If all other radio buttons are inside a `field` div, this one should be inside it too.

File: app/templates/components/forms/orders/order-form.hbs (hunk 2 above)
Comment (kushthedude) [reply app/templates/components/forms/orders/order-form.hbs]: I have added an image in this PR @CosmicCoder96 So it should appear

File: app/templates/components/forms/wizard/basic-details-step.hbs (hunk 4 above)
Comment (kushthedude) [reply app/templates/components/forms/wizard/basic-details-step.hbs]: So the user may get more information about `OMISE` through redirection to their gateway

Comment (kushthedude) [reply app/templates/components/forms/wizard/basic-details-step.hbs]: Will fix it

File: app/components/forms/wizard/basic-details-step.js (hunk 3 above)
Comment (kushthedude) [reply app/components/forms/wizard/basic-details-step.js]: Sorry , Making changes in dictionary

File: app/templates/components/forms/wizard/basic-details-step.hbs (hunk 4 above)
Comment (CosmicCoder96) [reply app/templates/components/forms/wizard/basic-details-step.hbs]: No I meant the `<span>` tag. Span tags without classes don't serve any purpose.

**Comment by kushthedude** [comment]
@CosmicCoder96 See now

File: app/templates/components/forms/wizard/basic-details-step.hbs [hunk 7]
```diff
@@ -308,14 +308,33 @@
           </div>
         {{/if}}
       {{/if}}
+      {{#if canAcceptOmise}}
+        <label>{{t 'Payment with Omise'}}</label>
+        <div class="field payments">
+          <div class="ui checkbox">
+            {{input type='checkbox' id='payment_by_omise' checked=data.event.canPayByOmise}}
+            <label for="payment_by_omise">
+              <span>{{t 'Yes, accept payment through Omise Gateway'}}</span>
```
Comment (CosmicCoder96) [inline app/templates/components/forms/wizard/basic-details-step.hbs]: Span tag is not required, the text string will do.

Comment (kushthedude) [reply app/templates/components/forms/wizard/basic-details-step.hbs]: Will push the commit

**Comment by kushthedude** [comment]
@CosmicCoder96 did changes

File: app/utils/dictionary/payment.js [hunk 8]
```diff
@@ -6,174 +6,199 @@ export const paymentCurrencies = [
     code   : 'PLN',
     symbol : 'zł',
     name   : 'Polish zloty',
-    stripe : true
+    stripe : true,
+    omise  : true
```
Comment (uds5501) [inline app/utils/dictionary/payment.js]: These are wrong. Please see omise's documentation. They have a very limited support in terms of currencies. Kindly use `omise: true` in the applicable positions only.

**Comment by kushthedude** [comment]
@CosmicCoder96 review

**Pull request merged**